lib/index.js

```diff
@@ -0,0 +1,166 @@
+/*
+ * Copyright (c) 2016 Digital Bazaar, Inc. All rights reserved.
+ */
+/* global webpage, window, document */
+var bedrock = require('bedrock');
+var cache = require('bedrock-redis');
+var config = bedrock.config;
+var express = require('bedrock-express').express;
+var phridge = require('phridge');
+var prerender = require('prerender-node');
+var querystring = require('querystring');
+var url = require('url');
+
+require('bedrock-requirejs');
+require('bedrock-views');
+
+var api = {};
+module.exports = api;
+
+require('./config');
+
+var logger = bedrock.loggers.get('app');
+var phantom = null;
+var requestQueue = {};
+
+var args = {
+  '--load-images': false,
+  '--ignore-ssl-errors': true,
+  '--ssl-protocol': 'tlsv1.2'
+};
+
+bedrock.events.on('bedrock.start', callback => {
+  // configure whitelist / blacklist
+  if(config.prerender.whitelist) {
+    prerender.whitelisted(config.prerender.whitelist);
+  }
+  if(config.prerender.blacklist) {
+    prerender.blacklisted(config.prerender.blacklist);
+  }
+  phridge.spawn(args).then(p => {
+    phantom = p;
+    return phantom.run(function() {
+      console.log('PhantomJS Started.');
+    });
+  }).then(() => callback(), callback);
+});
+
+bedrock.events.on('bedrock-express.configure.routes', (app, callback) => {
+  var routes = config.prerender.routes;
+  var routeKeys = Object.keys(config.prerender.routes);
+  if(routeKeys.length) {
+    var router = express.Router();
+    var error;
+    var success = routeKeys.every(route => {
+      if(typeof routes[route] !== 'object') {
+        error = new TypeError(
+          'Route options must be an object. Route: \'' + route +
+          '\' Value: \'' + routes[route] + '\'');
+        return false;
+      }
+      router.get(route, _prerender);
+      return true;
+    });
+    if(!success) {
+      return callback(error);
+    }
+    app.use(router);
+  }
+  callback();
+});
+
+function _prerender(req, res, next) {
+  if(!prerender.shouldShowPrerenderedPage(req)) {
+    return next();
```

Comment:
We want to go to the next middleware handler right, not the next route, right?

`next('route')` halts the current route and moves on to the next matching express route right?